File: docs/examples/01GetStart/src/main/scala/org/tresamigos/getstart/etl/Employment.scala
package org.tresamigos.getstart.etl

import org.tresamigos.getstart.core._
import org.tresamigos.smv._

/**
 * ETL Module Example
 */

object EmploymentRaw extends SmvModule("ETL Example: Employment") {

  override def version() = 0;
  override def requiresDS() = Seq(ExampleApp.employment);
 
  override def run(i: runParams) = {
    val srdd = i(ExampleApp.employment)
    import srdd.sqlContext.implicits._
    
    srdd.select(
      "EMP"
      )
  }
}

File: tools/templates/simple/src/main/python/stage1/employment.py
import smv
import pyspark.sql.functions as F

__all__ = ['EmploymentByState']

class Employment(smv.SmvCsvFile):
    def path(self):
        return "employment/CB1200CZ11.csv"

    def failAtParsingError(self):
        return False

class EmploymentByState(smv.SmvModule, smv.SmvOutput):
    """Python ETL Example: employ by state"""

    def requiresDS(self):
        return [Employment]

    def run(self, i):
        df = i[Employment]
        return df.groupBy(F.col("ST")).agg(F.sum(F.col("EMP")).alias("EMP"))
